### DISCUSSION TITLE
v3.2.1 - New features and some fixes

### DISCUSSION THREAD

**Discussion by Blackymas** [Blackymas/NSPanel_HA_Blueprint#545] [Announcements]
### As always, we would like to thank everyone who donated to this project and remind you that every donation helps to support this project - so if you haven't made a donation yet, now would be a good time ;)

[![Paypal](https://user-images.githubusercontent.com/41958506/212499642-b2fd097a-0938-4bfc-b37b-74df64592c58.png)](https://www.paypal.com/donate/?hosted_button_id=S974SWQMB8PB2)

&nbsp;

### Say hello to @edwardtfn  - we are happy to welcome you in the team :)

&nbsp;
&nbsp;

### In addition, we have added two new features.

&nbsp;

### The full release notes of v3.2 can be found here: https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v.3.2

&nbsp;

## Files that need to be reloaded:

1. nspanel_blueprint.yaml - v3.2.1
2. nspanel.tft - v3.2.1

> Esphome does not need to be updated if already running v3.2

&nbsp;

 _Update procedure can be read here:_

EN: https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-HowTo---All-important-thing-you-should-know

DE: https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(DE)-HowTo---Alle-wichtigen-Dinge-die-man-wissen-sollte

&nbsp;

## Community Updates
- Add Home Assistant icon page link to blueprint - thanks @bakito 

&nbsp;

## Overview of all changes

1. **_"Add new time and date format."_** - Now, different formats can be selected in the blueprint.

2. **_"Add RGB color picker"_** - We have added an "RGB Color Picker" to simplify the color selection.

3. **_"Add input_boolean domain for chips "_** - The input_boolean domain has been added for the chips

4. **_"Fix Notification Page"_** - Fix for both buttons on the notification page that did not work anymore

5. **_"Fix reload automation "_** - When the blueprint is modified or an automation is saved, the display does not reboot anymore

6. **_"Fix for slow response with "Loading...""_** - Fix the error where a missing/invalid entity causes the panel to slow down when opening a new page or keeps with "Loading..." on the screen until it timeouts.

8. **_"Fix climate on/off"_** - Fix error related to dict not containing on/off

9. **_"Fix for "Error rendering data template""_** - This fix the following error which happens when an entity with a non-numeric value ('unknown', 'unavailable', some text, etc.) is sent to the device

10. **_"Fix Nspanel stays on loading"_** 

11. **_"Fix for warning list object has no element n"_** - Fix for the following error happening when opening climate page

12. **_"Fix hw-button hold display"_** 

13. **_"Fix: remove hone page buttons "_** - The settings icon has been removed and replaced by a "hidden" button on the clock.

&nbsp;

### 1. Add new time and date format
Now, different formats can be selected in the blueprint.
![Bildschirmfoto 2023-03-26 um 11 57 41](https://user-images.githubusercontent.com/41958506/227768341-cba618dd-e23c-4f3b-9422-2f2aa42e6e0e.png)

&nbsp;

### 2. Add RGB color picker
We have added an "RGB Color Picker" to simplify the color selection.
![Bildschirmfoto 2023-03-26 um 11 58 08](https://user-images.githubusercontent.com/41958506/227768352-517f6379-e4ae-4c8b-9a29-3bca569340ad.png)

&nbsp;

## Next topics we are currently working on
See here: https://github.com/Blackymas/NSPanel_HA_Blueprint/labels/roadmap

## What's Changed
* Fix Notification by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/488
* Fix Notifications by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/502
* Simplifying code - Input area by @edwardtfn in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/482
* Optimizing Multilingual User Interface by @edwardtfn in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/498
* Removed incorrect instance of `- choice` by @edwardtfn in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/508
* Fix for slow response with "Loading..." by @edwardtfn in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/509
* upd: use rgb color-picker by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/516
* fix: Nspanel stays on loading by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/518
* Fix for "Error rendering data template" by @edwardtfn in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/514
* Fix error related to dict not containing on/off by @edwardtfn in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/513
* Support to date & time formats by @edwardtfn in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/510
* Add Home Assistant Icon Page to blueprint by @bakito in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/476
* Ignore IntelliJ IDEA project directory by @bakito in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/519
* Fix "Loading..." when selected before RGB picker by @edwardtfn in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/521
* Fix for warning "list object has no element n" by @edwardtfn in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/522
* Add support to `input_boolean` to chips by @edwardtfn in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/527
* upd: blueprint version 3.2.1 by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/528
* upd: reload automation triggers page refresh by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/529
* fix: hw-button hold display by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/530
* upd: remove unneeded link by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/531
* fix: remove hp-bottons on disable in blueprint by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/532
* v3.2.1 - new features and fixes by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/533
* upd: blank.tft by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/540
* fix: remove unneeded home.hotwater.pic by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/541
* v3.2.1 - new features and fixes by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/542
* v3.2.1 - new features and fixes by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/544

## New Contributors
* @bakito made their first contribution in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/476

**Full Changelog**: https://github.com/Blackymas/NSPanel_HA_Blueprint/compare/v.3.2...v3.2.1

<hr /><em>This discussion was created from the release <a href='https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v3.2.1'>v3.2.1 - New features and some fixes</a>.</em>

**Comment by lloydbayley** [comment]
Thank you again for another bigger and better version! :)

**Comment by jjnj023** [comment]
Wow, that looks like a great update!
Thanks to you I installed 6 panels. Just made a donation
I hope to install the new firmware over the weekend.
